### DISCUSSION TITLE
how to use .so files in .net maui project

### DISCUSSION THREAD

**Discussion by dgandhi17** [dotnet/maui#12279] [Q&A]
Have been trying to add .so file to resource lib folder setting nativelibrary as build action but it doesnt work
also not able to find any good documentation reference on how to use it with .net MAUI project structure.

**Comment by xtuzy** [comment]
Create a android class library to store .so, see https://github.com/xtuzy/Veldrid.SPIRV.NativeAsset/tree/master/Veldrid.SPIRV.AndroidNativeAsset

**Reply by dgandhi17** [reply]
So we get the .dll from individual native projects and reference it to .maui project?
inshort how do you suggest it to reference it to maui library project?

**Reply by xtuzy** [reply]
directly refer native project, or upload native project to nuget.org or github packages. native project only supply .so file, all dllimport can in a netstandard library

**Reply by dgandhi17** [reply]
if you have demo project available or can create it, other people can also benefit from it. Since there is no documentation of using it with .net maui. 

What i tried is from native project generated .dll and added as a reference to .net maui, the problem is the .dll got added to all the projects and also it didnt work.

**Comment by 406345** [comment]
so have you solved it? I have the same question. No documents can be found online, only https://learn.microsoft.com/en-us/xamarin/android/platform/native-libraries but it is for xamarin.
